dali/test/dali_plugin_manager_test.cc

```diff
@@ -18,61 +18,66 @@
 #include "dali/test/dali_test_utils.h"
 #include "dali/pipeline/pipeline.h"
 
-static const std::string& DummyPluginLibPath() {
+namespace other_ns {
+
+const std::string &DummyPluginLibPath() {
   static const std::string plugin_lib =
-      dali::test::CurrentExecutableDir() + "/libcustomdummyplugin.so";
+      dali::test::CurrentExecutableDir() + "/libdali_customdummyplugin.so";
   return plugin_lib;
 }
 
-static const std::string& DummyPluginLibPathGlobal() {
-  static const std::string plugin_lib = "libcustomdummyplugin.so";
+const std::string &DummyPluginLibPathGlobal() {
+  static const std::string plugin_lib = "libdali_customdummyplugin.so";
   return plugin_lib;
 }
 
-namespace other_ns {
-
-class DummyTest : public ::dali::DALITest {
- public:
-  static void LoadDummyPlugin() {
-    try {
-      ::dali::PluginManager::LoadLibrary(DummyPluginLibPath());
-    } catch(dali::DALIException &) {
-      ::dali::PluginManager::LoadLibrary(DummyPluginLibPathGlobal());
-    }
+void LoadDummyPlugin() {
+  try {
+    ::dali::PluginManager::LoadLibrary(DummyPluginLibPath());
+  } catch (dali::DALIException &) {
+    ::dali::PluginManager::LoadLibrary(DummyPluginLibPathGlobal());
   }
+}
 
-  void TestPlugin(const std::string &backend) {
-    LoadDummyPlugin();
+void TestPlugin(const std::string& backend) {
+  dali::TensorList<dali::CPUBackend> data;
+  dali::test::MakeRandomBatch(data, 3);
 
-    dali::TensorList<dali::CPUBackend> data;
-    dali::test::MakeRandomBatch(data, 3);
+  dali::Pipeline pipe(3, 1, 0);
+  pipe.AddExternalInput("data");
+  pipe.AddOperator(dali::OpSpec("CustomDummy")
+                       .AddArg("device", backend)
+                       .AddInput("data", backend)
+                       .AddOutput("out", backend));
+  std::vector<std::pair<std::string, std::string>> outputs = {{"out", backend}};
 
-    dali::Pipeline pipe(3, 1, 0);
-    pipe.AddExternalInput("data");
-    pipe.AddOperator(
-        dali::OpSpec("CustomDummy")
-        .AddArg("device", backend)
-        .AddInput("data", backend)
-        .AddOutput("out", backend));
-    std::vector<std::pair<std::string, std::string>> outputs = {{"out", backend}};
+  pipe.Build(outputs);
+  pipe.SetExternalInput("data", data);
+  dali::Workspace ws;
+  pipe.Run();
+  pipe.Outputs(&ws);
 
-    pipe.Build(outputs);
-    pipe.SetExternalInput("data", data);
-    dali::Workspace ws;
-    pipe.Run();
-    pipe.Outputs(&ws);
+  dali::test::CheckResults(ws, 3, 0, data);
 
-    dali::test::CheckResults(ws, 3, 0, data);
-  }
-};
+  _exit(0);  // This is to force each test to run in a separate process
+}
 
+TEST(DummyTest, TestPluginCPU) {
+  GTEST_FLAG_SET(death_test_style, "threadsafe");
+  LoadDummyPlugin();
+  EXPECT_EXIT(TestPlugin("cpu"), testing::ExitedWithCode(0), "");
```

Comment:
This is crucial. Before this change it was enough that the first testcase loads the plugin, as it was run in the same process. This change makes sure that each test case has to load the plugin

Reply:
Maybe it would be good to add a comment here to explain this

Reply:
Done